backend/routes/ClassData.go

```diff
@@ -58,32 +56,31 @@ type Subclass struct {
 	Description string `json:"description"`
 }
 
-// GetClassData is a pre-http.HandlerFunc for class data route ready to be a closure with *DatabaseStruct
+// GetClassData is a pre-http.HandlerFunc for class data route and will become a closure with *DatabaseStruct
 func GetClassData(writer http.ResponseWriter, request *http.Request, ds *db.DatabaseStruct) {
 
 	if request.Method != "POST" {
-		http.Error(writer, "Unsupported request type", http.StatusMethodNotAllowed)
+		errInvalidMethod(logTagClassData, writer, request)
 		return
 	}
 
 	body, err := ioutil.ReadAll(request.Body)
 
 	if err != nil {
-		http.Error(writer, "Error handling POST request", http.StatusBadRequest)
-		log.Printf("%s Failed to read request to %q from %q: %s", logTagClassData, request.RequestURI, request.RemoteAddr, err.Error())
+		errRequestBodyPost(logTagClassData, err, writer, request)
 		return
 	}
 
 	var classesToQuery []ClassDataRequest
 	err = json.Unmarshal(body, &classesToQuery)
 
 	if err != nil {
-		http.Error(writer, "Failed to parse request", http.StatusInternalServerError)
-		log.Printf("%s Failed to parse request to %q from %q: %s", logTagClassData, request.RequestURI, request.RemoteAddr, err.Error())
+		errRequestBodyParse(logTagClassData, err, writer, request)
 		return
 	}
 
 	// TODO: reduce to one query?
+
 	ret := make(map[string][]Subclass)
 	for i := 0; i < len(classesToQuery); i++ {
```

Comment:
Is there any particular concern why we do not combine all of them into one query?

Reply:
I suppose we could have a select from .. where course_num in (...), could be better